File: Mooga/src/de/lumpn/mooga/ranking/impl/CrowdingDistanceComparator.java
package de.lumpn.mooga.ranking.impl;

import java.util.Comparator;

/**
 * Compares individuals by total crowding distance.
 */
public final class CrowdingDistanceComparator implements
		Comparator<CrowdingDistanceIndividual> {

	@Override
	public int compare(CrowdingDistanceIndividual a, CrowdingDistanceIndividual b) {
		double distance1 = a.getTotalCrowdingDistance();
		double distance2 = b.getTotalCrowdingDistance();
		return Double.compare(distance1, distance2);
	}
}

File: csharp/InfiniteZelda/Mooga/CrowdingDistanceComparator.cs
using System.Collections.Generic;

namespace Lumpn.Mooga
{
    ///Compares individuals by crowding distance.
    public sealed class CrowdingDistanceComparator : IComparer<CrowdingDistanceIndividual>
    {
        private static readonly Comparer<double> comparer = Comparer<double>.Default;

        public int Compare(CrowdingDistanceIndividual a, CrowdingDistanceIndividual b)
        {
            return comparer.Compare(a.Distance, b.Distance);
        }
    }
}
